e2e_tests/suite/test_smoke.py

```diff
@@ -0,0 +1,114 @@
+import time
+
+import pytest
+import requests
+import json
+
+from botocore.waiter import WaiterModel, create_waiter_with_client
+
+from e2e_tests.settings import NGINX_API_VERSION, RECONFIGURATION_DELAY
+
+
+def wait_for_changes_in_api(req_url, desired_capacity) -> None:
+    resp = requests.get(req_url)
+    nginx_upstream = json.loads(resp.text)
+    counter = 0
+    while len(nginx_upstream['peers']) != desired_capacity and counter < 10:
+        time.sleep(RECONFIGURATION_DELAY)
+        counter = counter + 1
+        resp = requests.get(req_url)
+        nginx_upstream = json.loads(resp.text)
+
+
+def wait_for_changes_in_aws(autoscaling_client, group_name, desired_capacity) -> None:
+    waiter_name = 'autoscaling_completed'
+    argument = f"contains(AutoScalingGroups[?(starts_with(AutoScalingGroupName, `{group_name}`) == `true`)]." \
+               f"[length(Instances[?LifecycleState=='InService']) == `{desired_capacity}`][], `false`)"
```

Comment:
do we really need? is checking in `scale_aws_group` that the group has been scaled not enough?

Reply:
The only thing `scale_aws_group` does is setting DesiredCapacity to a new value. Instances doesn't start at once, we need to wait for them to become InService. I prefer to ensure that AWS part is done successfully before make an assertion in ASG tool. Otherwise it's hard to understand where the problem is.